client/peer.go

```diff
@@ -169,6 +226,53 @@ func (p *peer) Blocks(ctx context.Context, channel string, identity msp.SigningI
 	return bs.Blocks(), bs.Close, nil
 }
 
+func (p *peer) ParsedBlocks(ctx context.Context, channel string, identity msp.SigningIdentity, blockRange ...int64) (<-chan *block.Block, func() error, error) {
+	commonBlocks, commonCloser, err := p.Blocks(ctx, channel, identity, blockRange...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	parsedBlockChan := make(chan *block.Block)
+	go func() {
+		defer func() {
+			close(parsedBlockChan)
+		}()
+
+		for {
+			select {
+			case b, ok := <-commonBlocks:
+				if !ok {
+					return
+				}
+				if b == nil {
+					return
+				}
+
+				p.mu.Lock()
```

Comment:
Чтение из карты, если я не ошибаюсь c Golang версии 1.6, потоко безопасно.

```suggestion:-3+1
p.configBlocks[channel]
```